src/main/java/org/datadog/jmxfetch/util/MetadataHelper.java

```diff
@@ -1,19 +1,27 @@
 package org.datadog.jmxfetch.util;
 
 import java.io.IOException;
+import java.io.InputStream;
 import java.util.Properties;
 
 public class MetadataHelper {
     /**  Returns our own version number. */
     public static String getVersion() {
+        final Properties properties = new Properties();
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        InputStream stream = classLoader.getResourceAsStream("project.properties");
```

Comment:
I think this should be enough:

```
try (InputStream stream = classLoader.getResourceAsStream("project.properties")) {
     properties.load(stream);
     return properties.getProperty("version");
} 
```

Reply:
much better, thanks!

This requires Java 1.7, so I'll change the `maven.compiler` properties accordingly, we only support Java 1.7 and up in the JMXFetch releases shipped with v7/v6 agents anyway.